src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs

```diff
@@ -270,6 +273,13 @@ protected static T Cast<T>(object value)
             return currentDestination;
         }
 
+        private bool GeneratingEnum(IEnumerable<ISymbol> members)
+        {
+            // If we're generating an enum, every member will be an enum field
+            var field = members.FirstOrDefault() as IFieldSymbol;
```

Comment:
Is that always the case... Enum's never have anything else but fields?  Maybe ```.OfType<IFieldSymbol>().FirstOrDefault()``` in case?